cmd/proxy/main_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+const (
+	binName = "proxy"
+	timeout = 5 * time.Second
+)
+
+func waitForProxyLive(t *testing.T, c *http.Client) {
+	deadline := time.Now().Add(timeout)
+	for time.Now().Before(deadline) {
+		res, err := c.Get("http://martian.proxy/configure")
+		if err != nil {
+			time.Sleep(200 * time.Millisecond)
+			continue
+		}
+		if got, want := res.StatusCode, http.StatusOK; got != want {
+			t.Fatalf("GET config: got status code %d, want %d", got, want)
+		}
+		return
+	}
+	t.Fatalf("Proxy did not start up within %v seconds", timeout.Seconds())
+}
+
+func getProxiedClient(t *testing.T, proxyUrl string) *http.Client {
+	pu, err := url.Parse(proxyUrl)
+	if err != nil {
+		t.Fatalf("Parse proxy url: got error %v, want no error", err)
+	}
+	return &http.Client{Transport: &http.Transport{Proxy: http.ProxyURL(pu)}}
+}
+
+func TestProxyHttp(t *testing.T) {
```

Comment:
reame to `TestProxyEndToEnd` - "Http" doesn't really add any information, and I'd like to call this out as the end-to-end test.